geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala

```diff
@@ -119,7 +119,19 @@ class STIdxStrategy extends Strategy with Logging with IndexFilterHelpers {
     qp.copy(table = table, iterators = iterators, numThreads = numThreads, hasDuplicates = hasDupes)
   }
 
-  def getSTIIIterCfg(iteratorConfig: IteratorConfig,
+  def decomposeToGeometry(f: Filter): Seq[Geometry] = f match {
+    case bbox: BBOX =>
+      val bboxPoly = bbox.getExpression2.asInstanceOf[Literal].evaluate(null, classOf[Geometry])
+      Seq(bboxPoly)
+    case gf: BinarySpatialOperator =>
+      extractGeometry(gf)
+    case or: OrImpl =>
```

Comment:
currently other unit tests fail without this addition. I am not familiar with the STIdxStartegy so I don't know how this process worked in the past. The two unit tests I refer to fail because the previous mutations to the geometry filter fail to split on the "or" or fail to properly union the geometries. This sounds like a regression to me.